### DISCUSSION TITLE
Address Book fails after a password change

### DISCUSSION THREAD

**Discussion by bk-king1** [the-djmaze/snappymail#1644] [Q&A] [answered]
A user changed their password, and since then the address book does not show up in snappymail. Logs are showing:

Jun 27 09:58:15 webmail snappymail: [2024-06-26 23:58:15.467][e4e87017] Crypt[WARNING]: Decrypt() invalid $data or $key
Jun 27 09:58:15 webmail snappymail: [2024-06-26 23:58:15.468][e4e87017] AddressBook[ERROR]: CryptKeyError[111]

Where do I start? I can see the entries in the mySQL DB... but it does not contain any authentication data so wondering why the password change impacted this?

**Comment by cm-schl** [comment]
Do you use a ldap backend and the user changed its password there?
Maybe the problem could be connected with this: https://github.com/the-djmaze/snappymail/issues/1602#issuecomment-2135169643 although it would be strange that only the addressbook does not work and the login works perfectly. So keep that in mind before you try this 🙂

**Reply by bk-king1** [reply]
Snappymail and The email system in general are MySQL back ends.. but they are totally separate DBs. I cannot see any correlation between the 2. Snappymail just accesses the mailbox via IMAP. But yes, login works fine. It is just since the password change the AddressBook(which is configured within snappymail using its own DB) started failing. And only for this user.

If they login using just local-part the address book is there - albeit empty - and that is expected as snappymail will save the entries based on the username. so local-part and local-part@domain are different users in snappymails eyes. So something is wrong in the addressbook for this one user.

Wondering if the devs can point out what could raise those errors?

**Reply by cm-schl** [reply, marked as answer]
Hi,

whats the SnappyMail version you are using? Do you use SnappyMail standalone or inside Nextcloud?
The `CryptKeyError` "normally" comes up in situations where SnappyMail wants to look up the saved user password and can't decrypt it (because the password changed). So maybe this could be connected with some "ask for new password" popup that is not showing correctly or something similar.

You could try to save the whole user profile folder and then remove temporarily the .cryptkey file just to give some more insight if the problem is connected with the decryption of the file. When you change the authentication to `just local-part` normally a new user profile is generated (as you said) and therefore this could result to the same situation: the .cryptkey is regenerated and then it works again.

**Comment by bk-king1** [comment]
Awesome thank you!

V: 2.36.3
Deleted that .cryptkey file, got them to log back in and all is good in the world. Address book has returned.

Thanks again!
